sql/drop.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: XisiHuang (cockhuangxh@163.com)
+
+package sql
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/keys"
+	"github.com/cockroachdb/cockroach/sql/parser"
+)
+
+// DropTable drops a table.
+func (p *planner) DropTable(n *parser.DropTable) (planNode, error) {
+	// TODO(XisiHuang): should do truncate and delete descriptor in
+	// the same txn
+	for i, tableQualifiedName := range n.Names {
+		if err := p.normalizeTableName(tableQualifiedName); err != nil {
+			return nil, err
+		}
+
+		dbDesc, err := p.getDatabaseDesc(tableQualifiedName.Database())
+		if err != nil {
+			return nil, err
+		}
+
+		nameKey := keys.MakeNameMetadataKey(dbDesc.ID, tableQualifiedName.Table())
+		gr, err := p.db.Get(nameKey)
+		if err != nil {
+			return nil, err
+		}
+
+		if !gr.Exists() {
+			if n.IfExists {
+				// Noop.
+				continue
+			}
+			// Key does not exists, but we want it to: error out.
+			return nil, fmt.Errorf("TableDescriptor \"%s\" does not exist", nameKey.String())
+		}
+
+		_, err = p.Truncate(&parser.Truncate{Tables: n.Names[i : i+1]})
```

Comment:
can you change this to the shadowing form here and below? `if _, err := ....; err != nil { ..`